spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/StringUtil.java

```diff
@@ -21,21 +21,114 @@ public final class StringUtil {
+    public static final int DEFAULT_RANDOM_STRING_LENGTH = 8;
+
+    /** Character set consisting of upper- and lower-case alphabetical ASCII characters */
+    public static final String CHARSET_ALPHABETICAL = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+    /** Character set consisting of numeric ASCII characters */
+    public static final String CHARSET_NUMERIC = "0123456789";
+
+    /** Character set consisting of characters representing base-16 (hexidecimal) values (0-9, A-F) */
+    public static final String CHARSET_NUMERIC_HEX = "0123456789ABCDEF";
+
+    /** Character set consisting of numeric and upper- and lower-case alphabetical ASCII characters */
+    public static final String CHARSET_ALPHANUMERIC = CHARSET_ALPHABETICAL + CHARSET_NUMERIC;
+
     private StringUtil() {
         throw new UnsupportedOperationException();
     }
 
     /**
-     * Generates a random suffix of 7 characters or fewer.
+     * Generates a random string of characters from the specified charset string.
+     *
+     * @param length
+     *  the length of the string to generate; must be a positive integer
+     *
+     * @param charset
+     *  a string representing the character set to use for generating the string
+     *
+     * @throws IllegalArgumentException
+     *  if length is a non-positive integer, or charset is null or empty
+     *
+     * @return
+     *  a randomly generated string using the characters from the specified charset string
+     */
+    public static String random(int length, String charset) {
+        if (length < 1) {
+            throw new IllegalArgumentException("length is a non-positive integer");
+        }
+
+        if (charset == null || charset.isEmpty()) {
+            throw new IllegalArgumentException("charset is null or empty");
+        }
+
+        return RANDOM.ints(length, 0, charset.length())
+            .map(charset::charAt)
+            .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+            .toString();
+    }
+
+    /**
+     * Generates a string of characters from the specified charset string, appended to the given
+     * prefix string. If the prefix is null or empty, the resultant string will only contain the
+     * generated portion.
+     *
+     * @param prefix
+     *  the prefix to prepend to the generated string
+     *
+     * @param length
+     *  the length of the string to generate, not counting the prefix; must be a positive integer
+     *
+     * @param charset
+     *  a string representing the character set to use for generating the string
+     *
+     * @throws IllegalArgumentException
+     *  if length is a non-positive integer, or charset is null or empty
+     *
+     * @return
+     *  a randomly generated string using the characters from the specified charset string appended
+     *  to the provided prefix
+     */
+    public static String random(String prefix, int length, String charset) {
```

Comment:
Do we want to add the hyphen between the prefix and suffix automatically so that the caller doesn't always have to always add it to their prefix? I guess that would give the caller less control over creating a random string exactly how they want it, but it would mean our random strings would be uniform.

Reply:
I went back and forth on this myself, and landed on not including it, since it (a) mucks with the number of characters output by the string in a way that's not good for our random space problem and (b) doesn't make sense for every format of string, and it's probably clearer overall to have the caller specify.

Reply:
Both points make a lot of sense. I'm good with the way you implemented it!